### ISSUE TITLE
Debian_10 and Ubuntu_18.04 packages on OBS

### ISSUE THREAD

**Issue by munix9** [CelestiaProject/Celestia#388]
The OBS now also creates packages for Debian_10 and Ubuntu_18.04, as well as for openSUSE Leap 15.1 and Tumbleweed (git master).

https://build.opensuse.org/package/show/home:munix9:unstable/celestia
https://software.opensuse.org/download.html?project=home:munix9:unstable&package=celestia

https://build.opensuse.org/package/binaries/home:munix9:unstable/celestia/Debian_10
https://build.opensuse.org/package/binaries/home:munix9:unstable/celestia/Ubuntu_18.04

Some small adjustments were necessary:

- OBS needs an initial ```celestia.dsc```. I have created one
https://build.opensuse.org/package/view_file/home:munix9:unstable/celestia/celestia.dsc?expand=1

- a patch for using GIT_COMMIT in the build
https://build.opensuse.org/package/view_file/home:munix9:unstable/celestia/debian-fix-GIT_COMMIT.patch?expand=1
It's just a little hack. A concrete solution should consider the implementation of GIT_COMMIT in the ```CMakeLists.txt```.

Enabling Spice would also be possible, similar to what I've already done for the openSUSE packages:
https://build.opensuse.org/package/view_file/home:munix9:unstable/celestia/fix-spice-config.patch?expand=1
The configuration for the shared library ```libspice65.so``` is defined in ```/etc/ld.so.conf.d/celestia.conf```:
```
$ cat /etc/ld.so.conf.d/celestia.conf
/usr/lib64/celestia
```
Sorry, currently no PR available, I am constantly on the go :-\

More information about OBS and Debian/Ubuntu can be found at
https://en.opensuse.org/openSUSE:Build_Service_Debian_builds

Please test and give feedback (@375gnu) as I don't use Debian/Ubuntu natively myself (and my VM disk is bursting at the seams).

**Comment by 375gnu** [comment]
Thank you! I will test it.

**Comment by 375gnu** [comment]
To enable repository:
```
curl https://download.opensuse.org/repositories/home:/munix9:/unstable/Debian_10/Release.key | sudo apt-key add -
echo "https://download.opensuse.org/repositories/home:/munix9:/unstable/Debian_10/ ./" | sudo tee /etc/apt/sources.list.d/celestia-obs.list

sudo apt update && sudo apt install celestia
```

**Comment by 375gnu** [comment]
I also need to fix dependencies for those who have old 1.6.1 installed from Debian 8

**Comment by 375gnu** [comment]
@munix9 is it possible to change version representation? Now it looks like `1.7.0+git.1567624803.31cc7363d+182.1` so problems are:
1. `+` sign make it higher then just `1.7.0` or `1.7.0-n`. Is it possible to replace it with `~`?
1. date is written as Unix timestamp. Is it possible to replace it with %Y%m%d?

**Comment by munix9** [comment]
eg: `1.7.0~git.20190904.31cc7363d+182.1` ?

**Comment by 375gnu** [comment]
Yes, this will be much better. But, `1.7.0~git20190904+31cc7363d-182.1` would be even better as it follows usual debian convention.

**Comment by munix9** [comment]
hm, I would prefer
`1.7.0~git20190904+31cc7363d+182.1`

The "+" sign for the release is added by debtransform in OBS
https://github.com/openSUSE/obs-build/blob/master/debtransform#L374-L385
Strangely enough, a "-" character was apparently used earlier and then changed to a "+" character, see https://github.com/openSUSE/obs-build/pull/352

Alternatively the config `DEBTRANSFORM-RELEASE` for the Debian/Ubuntu build can be removed, then the result should look like
`1.7.0~git20190904+31cc7363d`

**Comment by 375gnu** [comment]
> Alternatively the config `DEBTRANSFORM-RELEASE` for the Debian/Ubuntu build can be removed, then the result should look like
> `1.7.0~git20190904+31cc7363d`

Yes please. As we build "native" packages then we don't need release.

**Comment by munix9** [comment]
done.

**Comment by 375gnu** [comment]
Thank you! Looks good!

**Comment by 375gnu** [comment]
closing this as everything works fine now

**Comment by SDRausty** [comment]
> closing this as everything works fine now

```
root@localhost:~# uname -oms
Linux aarch64 GNU/Linux
root@localhost:~# cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux bullseye/sid"
NAME="Debian GNU/Linux"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

After reviewing https://packages.debian.org/jessie/celestia and https://wiki.debian.org/it/EduCelestia and searching ` debian install celestia ` on the Internet: 
```
root@localhost:~# apt-get install celestia
Reading package lists... Done
Building dependency tree
Reading state information... Done
Package celestia is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'celestia' has no installation candidate
root@localhost:~# cat /etc/apt/sources.list
deb https://deb.debian.org/debian unstable main
```

The results with stable were the same as with unstable.  Debian runs, but Celestia is absent;  @375gnu do you have any advice about getting Celestia installed and running in my Android smartphone.   Compiling as well as installing packages are open options on this Android device.

**Comment by 375gnu** [comment]
Hi @SDRausty,

Currently Debian doesn't have Celestia packaged. If you want it please use OBS repository as stated above:
```
curl https://download.opensuse.org/repositories/home:/munix9:/unstable/Debian_10/Release.key | sudo apt-key add -
echo "deb https://download.opensuse.org/repositories/home:/munix9:/unstable/Debian_10/ ./" | sudo tee /etc/apt/sources.list.d/celestia-obs.list

sudo apt update && sudo apt install celestia
```

But we don't have `arm64` builds yet. You could try to compile it yourself. But still I'm not sure it will run properly even in chroot because Celestia requires desktop GL 2.1 while android devices provide OpenGL ES, so you will software (= too slow) rendering. Or you could try to experiment with GLshim library which implements GL on GL ES.

**Comment by SDRausty** [comment]
Hello @375gnu,

I managed to compile Celestia on handheld in Debian Termux PRoot.  It appears that the build went okay.  I did not manage to test it in xserver on device.

I was hoping to stay with the command line.  But it seems that this has been depreciated.  Is running Celestia only from the command line still possible?

For example: plot trajectory and plot views of objects to a file and render images from this file into Full HD Video?

**Comment by 375gnu** [comment]
@SDRausty no, such mode is not supported. Maybe with offscreen Mesa might help you, but I haven't used it.
